src/topology.rs

```diff
@@ -9,6 +9,7 @@ pub struct Topology {
     nodes: Vec<NodeId>,                  // The list of nodes in the topology
     edges: HashMap<NodeId, HashSet<NodeId>>, // All the connections between nodes.
     labels: HashMap<NodeId, String>,     // The labels of the nodes
+    node_types: HashMap<NodeId, String>,      // The types of the nodes
```

Comment:
And for this new map, when adding a node we insert the type of that node? (And ourselves + the neighbors drones)?

Reply:
For each new node you add the node type, the label is for the controller, used exclusively when loading up the initial topology of the network with all the members